File: app/src/main/java/com/alvinhkh/buseta/model/EmptyElementConverter.java
package com.alvinhkh.buseta.model;

import org.simpleframework.xml.convert.Converter;
import org.simpleframework.xml.stream.InputNode;
import org.simpleframework.xml.stream.OutputNode;

/*
 * https://stackoverflow.com/a/17220939/2411672
 */
public class EmptyElementConverter implements Converter<String> {
    @Override
    public String read(InputNode node) throws Exception
    {
        /* Implement if required */
        throw new UnsupportedOperationException("Not supported yet.");
    }

    @Override
    public void write(OutputNode node, String value) throws Exception
    {
        /* Simple implementation: do nothing here ;-) */
    }
}

File: app/src/main/java/com/alvinhkh/buseta/mtr/dao/EmptyElementConverter.kt
package com.alvinhkh.buseta.mtr.dao

import org.simpleframework.xml.convert.Converter
import org.simpleframework.xml.stream.InputNode
import org.simpleframework.xml.stream.OutputNode

/*
 * https://stackoverflow.com/a/17220939/2411672
 */
class EmptyElementConverter<T> : Converter<T> {
    @Throws(Exception::class)
    override fun read(node: InputNode): T {
        /* Implement if required */
        throw UnsupportedOperationException("Not supported yet.")
    }

    @Throws(Exception::class)
    override fun write(node: OutputNode, value: T) {
        /* Simple implementation: do nothing here ;-) */
    }
}